src/components/RegisterController/RegisterController.js

```diff
@@ -0,0 +1,269 @@
+import { useState } from "react";
+import { useSelector, useDispatch, useStore } from "react-redux";
+import { useHistory } from "react-router-dom";
+
+import { getBakery } from "app/selectors";
+import { connectAndStartPolling } from "app/actions";
+
+import { useQueryParam, StringParam } from "use-query-params";
+
+import classNames from "classnames";
+
+import useLocalStorage from "hooks/useLocalStorage";
+
+import SlideInOut from "animations/SlideInOut";
+
+import Aside from "components/Aside/Aside";
+
+import "./register-controller.scss";
+
+export default function RegisterController() {
+  const [formValues, setFormValues] = useState({});
+  const dispatch = useDispatch();
+  const reduxStore = useStore();
+  const bakery = useSelector(getBakery);
+  const [additionalControllers, setAdditionalControllers] = useLocalStorage(
+    "additionalControllers",
+    []
+  );
+  const history = useHistory();
+
+  function closePanel() {
+    history.push("/controllers");
+  }
+
+  function handleRegisterAController(e) {
+    e.preventDefault();
+    // XXX Validate form values
+    additionalControllers.push([
+      `wss://${formValues.wsControllerHost}/api`, // wsControllerURL
+      { user: formValues.username, password: formValues.password }, // credentials
+      null, // bakery
+      formValues.identityProviderAvailable, // identityProviderAvailable
+      true, // additional controller
+    ]);
+    setAdditionalControllers(additionalControllers);
+    dispatch(connectAndStartPolling(reduxStore, bakery));
+    closePanel(); // Close the SlidePanel
+  }
+
+  function handleInputChange(e) {
+    const newFormValues = { ...formValues };
+    newFormValues[e.target.name] =
+      e.target.type === "checkbox" ? e.target.checked : e.target.value;
+    setFormValues(newFormValues);
+  }
+
+  function generateTheControllerLink(controllerIP) {
+    if (!controllerIP) {
+      return "the controller";
+    }
+    const dashboardLink = `https://${controllerIP}/dashboard`;
+    return (
+      <a href={dashboardLink} target="_blank" rel="noopener noreferrer">
+        the controller
+      </a>
+    );
+  }
+
+  const [panelQs, setPanelQs] = useQueryParam("panel", StringParam);
+
+  return (
+    <Aside>
+      <SlideInOut distanceInPx={725}>
```

Comment:
Should the sliding be in the `Aside` instead of the children of the aside? I feel like if I want to put content in a panel I'd just wrap it in the `Aside` and it should take care of the rest.